src/common/iceberg.cpp

```diff
@@ -46,9 +46,22 @@ IcebergTable IcebergTable::Load(const string &iceberg_path, IcebergSnapshot &sna
 	return ret;
 }
 
-unique_ptr<SnapshotParseInfo> IcebergSnapshot::GetParseInfo(yyjson_doc &metadata_json) {
-	SnapshotParseInfo info {};
-	auto root = yyjson_doc_get_root(&metadata_json);
+unique_ptr<IcebergMetadata> IcebergMetadata::Parse(const string &path, FileSystem &fs,
```

Comment:
This includes changes merged into `main`, we could merge `main` into `feature` again to remove these changes from this diff, if that's preferable?